### PR TITLE
Refactored flags for tunnel related options

### PR DESCRIPTION
**Pull request by gberl002**
Restructured tests
Added tests for tunnel related flags

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+
+- Ziti CLI
+  - Added CLI flags for setting router tunneler capability
+
+## Ziti CLI
+### Added CLI flags for setting router tunneler capability
+Ziti CLI `ziti create config router edge` now has three new flags
+#### --disableTunneler
+The tunneler capability for routers is enabled by default, the `disableTunneler` flag can now be used to disable the 
+tunneler capability in the router config.
```
Comment (dovholuknf) [inline CHANGELOG.md:13]: what happens when i supply --disableTunneler and then set the mode/lanIf? that should be called out here imo

Comment (dovholuknf) [reply CHANGELOG.md:13]: actaully - feels like maybe --disableTunneler should just be removed and three 'modes' supported: none, host, tproxy

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+modes. The default tunneler mode is `host` mode, choosing `none` will disable tunnel capabilities for the router.
+
+Examples:
+```shell
+ziti create config router edge --routerName myRouter --tunnelerMode tproxy
+
+ziti create config router edge --routerName myRouter --tunnelerMode none
+```
+#### --lanInterface
+If using the `tproxy` tunneler mode, there is an optional `lanIf` section in the config to identify an interface to use.
```
Comment (dovholuknf) [inline CHANGELOG.md:22]: what happens with `tunnelerMode: none` and `lanInterface` is specfified?

Comment (gberl002) [reply CHANGELOG.md:22]: Nothing, the lanIf only shows up in the config if tproxy is set as the mode.

Comment (gberl002) [reply CHANGELOG.md:22]: Hmmm, actually I don't like what happened here, I'm going to update this. I don't like the blank lines commented out and I don't like that it says `mode: none`. Obviously they don't affect the outcome but it just seems, unrefined to me.

```
listeners:
# bindings of edge and tunnel requires an "edge" section below
  - binding: edge
    address: tls:0.0.0.0:3022
    options:
      advertise: MacBook-Pro.local:3022
      connectTimeoutMs: 1000
      getSessionTimeout: 60
#  - binding: tunnel
#    options:
#      mode: none #tproxy|host
#
#
```

Comment (dovholuknf) [reply CHANGELOG.md:22]: so are we waiting for a new round of changes? regardless I think these are fine to merge into the other branch

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**